### ISSUE TITLE
oAuth popup window doesn't close. 

### ISSUE THREAD

**Issue by rrubo** [alexandrtovmach/react-twitter-login#8]
Hi, I'm not sure whether this is a supposed behavior, but I've implemented your component like this:

```
<TwitterLogin
          authCallback={this.authHandler}
          consumerKey={twitterConsumerKey}
          consumerSecret={twitterConsumerSecret}
          callbackUrl={twitterCallbackUrl}
        />
```
```
authHandler = (err, data) => {
    console.log(err, data);
  };
```

On click, it does open the popup window, and after Twitter authentication it returns the response token. However, the popup window stays open, and I don't see `data` logged in console. 

Could you advise?

**Comment by alexandrtovmach** [comment]
@rrubo Thank you for report. It seems like edge case issue, could you try with different environments (Incognito mode, FF/Chrome), and provide any additional data like logs etc. Also, reproducible example will be very useful =)

In addition, test in the same environment demo example: https://alexandrtovmach.github.io/react-twitter-login/

**Comment by razvanilin** [comment]
I noticed the same problem. Basically what happens is that the callback URL is accessed in the popup window instead of redirecting in the main page. The `authHandler` gets called when the window is closed manually.

Tested in Chrome Version 80.0.3987.149 and Firefox 74.0 with the minimal implementation in the repo's README

The console logs after closing the popup manually:
```
err => User closed OAuth popup
data => undefined
```

**Comment by highsmallxu** [comment]
Hi, I'm having the same issue, is there any update on this?

**Comment by alexandrtovmach** [comment]
I'm busy now, so PR with solution are welcome 😉

**Comment by alexandrtovmach** [comment]
@highsmallxu Could you provide more details what exactly happens. Code example, logs, network logs etc.

**Comment by ronny1020** [comment]
I had the same issue at first. However, I  found the reason which may cause this issue in my case.

For local testing, We know the callback URL of Twitter has to use "127.0.0.1" instead of "localhost".
So, if visiting the page by "localhost", the URL of the popup window and the main window would be different. it would cause this issue.

In my case, visiting the page by "127.0.0.1" would be fixed.
I hope this can help.

**Comment by aneeshpanoli** [comment]
From my testing, this problem still exists. However if the `redirect url `and `window URL` are the same, everything works as expected.

**Comment by 1511372** [comment]
> From my testing, this problem still exists. However if the `redirect url `and `window URL` are the same, everything works as expected.

Thanks, @aneeshpanoli, it worked perfectly.

**Comment by alexandrtovmach** [comment]
@aneeshpanoli Thank you for good suggestion!
Yeah, it seems like a point that I missed in documentation ― `callbackUrl` return to the same page. It's expected behavior, because after redirect package should parse query parameters, send one more additional request and return data.

Could everyone with this issue confirm ― issue can be reproduced ONLY for cases when `callbackUrl !== window.location.href`?

**Comment by alexandrtovmach** [comment]
I'm just thinking :thinking:, maybe better to remove `callbackUrl` property at all, because it's SHOULD BE equal to `window.location.href`

**Comment by alexandrtovmach** [comment]
Should be fixed in [v1.2.1](https://www.npmjs.com/package/react-twitter-login/v/1.2.1).
Changes:
- `callbackUrl` deprecated and hardcoded with `window.location.href`
- added notes to README
- added deprecation notes and warnings
